src/test/java/com/quorum/gauge/PrivateSmartContract.java

```diff
@@ -422,4 +422,33 @@ public void sendPrivateSmartContracts(int count, QuorumNode source, String targe
         BigInteger blockNumber = Observable.zip(allObservableContracts, args -> utilService.getCurrentBlockNumber().blockingFirst()).blockingFirst().getBlockNumber();
         assertThat(blockNumber).isNotEqualTo(currentBlockNumber());
     }
+
+    @Step("Send <count> simple private raw smart contracts from wallet <wallet> in <source> and it's separately private for <targets> and saved under <store>")
+    public void sendRawPrivateSmartContracts(int count, WalletData wallet, QuorumNode source, String targets, String store) {
+        String[] target = targets.split(",");
+        QuorumNode[] targetNodes = new QuorumNode[target.length];
+        for (int i = 0; i < targetNodes.length; i++) {
+            targetNodes[i] = QuorumNode.valueOf(target[i]);
+        }
+
+        try {
+            List deployedContracts = Observable.zip(
+                rawContractService.createNRawSimplePrivateContract(count, wallet, source, targetNodes),
+                (a) -> a
+            ).collectInto(new ArrayList<>(), (objects, deployedContract) -> objects.addAll(Arrays.asList(deployedContract))).subscribeOn(Schedulers.io()).blockingGet();
+            DataStoreFactory.getScenarioDataStore().put(store, deployedContracts);
+        } catch (Exception e) {
+            Assert.fail();
+        }
+    }
+
+    @Step("Verify the stored data <store> into the store to have the random value")
```

Comment:
`to have the the right value` or something similar. we know that is random, but the step checks that the deployed has the right value comparing to the used in the constructor.

Reply:
updated